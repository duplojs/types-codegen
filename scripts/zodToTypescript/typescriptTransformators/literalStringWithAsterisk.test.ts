import { ZodToTypescript } from "@duplojs/zod-to-typescript";
import { zod, ZodSpace } from "@duplojs/core";
import { literalStringWithAsterisk } from "./literalStringWithAsterisk";
import { ZodLiteral } from "zod";

it("literalStringWithAsterisk", () => {
	const zodToTypescript = new ZodToTypescript([literalStringWithAsterisk]);

	zodToTypescript.append(
		new ZodLiteral({
			value: "totot*test*ff",
			typeName: ZodSpace.ZodFirstPartyTypeKind.ZodLiteral,
			asteriskToStringInterpolation: true,
		}),
		"Test",
	);

	zodToTypescript.append(
		new ZodLiteral({
			value: "/users/*",
			typeName: ZodSpace.ZodFirstPartyTypeKind.ZodLiteral,
			asteriskToStringInterpolation: true,
		}),
		"Path",
	);

	expect(
		zodToTypescript.toString(true),
	).toMatchSnapshot();
});
