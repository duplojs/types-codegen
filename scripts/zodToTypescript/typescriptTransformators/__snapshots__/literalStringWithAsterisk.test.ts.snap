// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`literalStringWithAsterisk 1`] = `
"type Test = \`totot\${string}test\${string}ff\`;

export { Test };

type Path = \`/users/\${string}\`;

export { Path };"
`;
